debezium-core/src/main/java/io/debezium/pipeline/ChangeEventSourceCoordinator.java

```diff
@@ -240,7 +240,7 @@ public void doBlockingSnapshot(P partition, OffsetContext offsetContext, Snapsho
             streaming = true;
 
             try {
-
+                streaming = snapshotterService.getSnapshotter().shouldStream();
```

Comment:
I think we can just use `snapshotterService.getSnapshotter().shouldStream()` to initialize the `streaming` variable so that the blocking snapshot will effectively wait only if the snapshot mode support streaming.